### PR TITLE
Initial commit of Dockerfile for Fedora 31

### PR DESCRIPTION
**Pull request by tachoknight**
I'm starting with Fedora 31 specifically because it uses Clang 9 which I know will build Swift from master.

### CODE REVIEW

**Comment by tomerd** [comment]
nice one! @shahmishal what command would trigger testing this new file?

File: swift-ci/master/fedora/31/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	ninja-build 				\
+	python-unversioned-command 	\
+	python27 					\
+	python3 					\
+	python3-devel 				\
+	python3-distro 				\
+	python3-six 				\
+	rsync 						\
+	swig 
+
```
Comment (tomerd) [inline swift-ci/master/fedora/31/Dockerfile:28]: nit: align backslashes

**Comment by tomerd** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
Please test will pick up the new dockerfile.

File: swift-ci/master/fedora/31/Dockerfile (hunk 1 above)
Comment (tachoknight) [reply swift-ci/master/fedora/31/Dockerfile:28]: Arg, this has been a problem with having different `.vimrc` files on different machines. I'll fix it.

**Comment by tomerd** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
@shahmishal I cant's find the results for the fedora image added here in the CI logs, can you confirm it was picked up?

**Comment by shahmishal** [comment]
@tomerd It did not pick up the change. I have fixed the issue with pull request testing system. 

Starting new build: 

@swift-ci Please test

**Comment by shahmishal** [comment]
Build passed - https://ci.swift.org/job/swift-docker-PR-linux/88/artifact/PASSED_swift-ci_master_fedora_31.log

**Review by shahmishal — APPROVED** [review]

**Pull request merged**